### ISSUE TITLE
Character '<' does not encode to &lt; in the FhirSerializer

### ISSUE THREAD

**Issue by marcovisserFurore** [ewoutkramer/fhir-net-api#531]
It looks like the FhirSerializer.SerializeResourceToXml will encode > sign to &gt; but when < is passed we get an error. 
![image](https://user-images.githubusercontent.com/31658535/35522765-7e134b12-051d-11e8-8d43-a42fc25e3118.png)

**Comment by brianpos** [comment]
This isn't an issue of the serializer, that is your content.
The narrative os xhtml, and what you are showing there is invalid xhtml.
The error is correct.

**Comment by marcovisserFurore** [comment]
You are right Brian. I investigated this issue yesterday and I came up with the same conclusion. I gave the user the following response:

I have checked the FHIR standard and the element text is of type Narrative (https://www.hl7.org/fhir/narrative.html#Narrative). This element consists of a status and div element. This div element must consist of limited xhtml content. I quote: 
  

> The contents of the div element are an XHTML fragment that SHALL contain only the basic HTML formatting elements described in chapters 7-11 (except section 4 of chapter 9) and 15 of the HTML 4.0 standard, <a> elements (either name or href), images and internally contained style attributes. The XHTML content SHALL NOT contain a head, a body element, external stylesheet references, deprecated elements, scripts, forms, base/link/xlink, frames, iframes, objects or event related attributes (e.g. onClick). This is to ensure that the content of the narrative is contained within the resource and that there is no active content. Such content would introduce security issues and potentially safety issues with regard to extracting text from the XHTML. 
  

And 

> Note that the XHTML is contained in general XML so there is no support for HTML entities like &nbsp; or &copy; etc. Unicode characters SHALL be used instead. Unicode &#160; substitutes for &nbsp;. 
  

The ‘<’ should be avoided because it conflicts with the beginning of a tag. I quote the HTML 4.0 standard (https://www.w3.org/TR/1998/REC-html40-19980424/charset.html#h-5.3.2) : 

> Four character entity references deserve special mention since they are frequently used to escape special characters: 
> ·	"&lt;" represents the < sign. 
> ·	"&gt;" represents the > sign. 
> ·	"&amp;" represents the & sign. 
> ·	"&quot; represents the " mark. 
> Authors wishing to put the "<" character in text should use "&lt;" (ASCII decimal 60) to avoid possible confusion with the beginning of a tag (start tag open delimiter). Similarly, authors should use "&gt;" (ASCII decimal 62) in text instead of ">" to avoid problems with older user agents that incorrectly perceive this as the end of a tag (tag close delimiter) when it appears in quoted attribute values. 
> 

So to summarize: 
Don’t use the ‘<’ and ‘>’ characters inside the div element, because it is not valid xhtml and thus is not valid according to the FHIR standard. The “>” character is processed correctly by the API, because it could not find an opening “<” character and it translates it automatically to a &gt; element. So see this a bonus, but actually you should avoid the ‘>’ characters. 
  
I hope I clarified it for you. Because it’s not according to the FHIR standard, there will be no change for the API. I would to advise you to encode the < and > characters before the serialization.
